### PR TITLE
Simplify and consolidate release specific docs

### PR DESCRIPTION
**Pull request by afrittoli**
# Changes

The list of release-specific docs and examples in the README it was becoming too long to be useful and readable. Besides, the only version of the docs that is useful for each minor release is those associated with the latest patch since patches only introduce security, regression and documentation fixes.

In the main README only include the link to the docs @ HEAD, and a link to the releases page.

On the releases page, for each latest patch version, add links to docs and examples.

Update the release cheatsheet accordingly.

Signed-off-by: Andrea Frittoli <andrea.frittoli@uk.ibm.com>

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

/kind documentation

/cc @lbernick

### CODE REVIEW

File: tekton/release-cheat-sheet.md [hunk 1]
```diff
@@ -144,11 +144,20 @@ the pipelines repo, a terminal window and a text editor.
    (This can be done on the Github UI.)
    Make sure to fetch the commit specified in `TEKTON_RELEASE_GIT_SHA` to create the released branch.
 
-1. Edit `README.md` on `main` branch, add entry to docs table with latest release links. 
+1. If the release introduces a new minimum version of Kubernets required,
```
Comment (lbernick) [inline tekton/release-cheat-sheet.md:147]:
```suggestion
1. If the release introduces a new minimum version of Kubernetes required,
```

**Review by lbernick — APPROVED** [review]
thanks Andrea!

File: tekton/release-cheat-sheet.md (hunk 1 above)
Comment (afrittoli) [reply tekton/release-cheat-sheet.md]: Thanks, fixed

**Review by jerop — APPROVED** [review]
/lgtm

**Comment by jerop** [comment]
/retest

another cloud events flake - https://github.com/tektoncd/pipeline/issues/5160

**Pull request merged**